conversation_classification/kaggle/model.py

```diff
@@ -0,0 +1,307 @@
+
+    x_train = np.array(list(vocab_processor.fit_transform(x_train_text)))
+    x_test = np.array(list(vocab_processor.fit_transform(x_test_text)))
+    y_train = np.array(y_train)
+    y_test = np.array(y_test)
+
+    n_words = len(vocab_processor.vocabulary_)
+    tf.logging.info('Total words: %d' % n_words)
+
+    # Build model
+    if FLAGS.model == 'bag_of_words':
+      model_fn = bag_of_words_model
+
+      # Subtract 1 because VocabularyProcessor outputs a word-id matrix where word
+      # ids start from 1 and 0 means 'no word'. But categorical_column_with_identity
+      # assumes 0-based count and uses -1 for missing word.
+      x_train -= 1
+      x_test -= 1
+    else:
+      tf.logging.error("Unknown specified model '{}', must be one of {}"
+                       .format(FLAGS.model, MODEL_LIST))
+      raise ValueError
+
+    classifier = tf.estimator.Estimator(
+      model_fn=model_fn,
+      config=tf.contrib.learn.RunConfig(
+        tf_random_seed=TRAIN_SEED,
+      ),
+      model_dir=None)
+
+    # Train model
+    train_input_fn = tf.estimator.inputs.numpy_input_fn(
+      x={WORDS_FEATURE: x_train},
+      y=y_train,
+      batch_size=BATCH_SIZE,
+      num_epochs=None, # Note: For training, set this to None, so the input_fn
+                       # keeps returning data until the required number of train
+                       # steps is reached.
+      shuffle=True)
+
+    classifier.train(input_fn=train_input_fn, steps=TRAIN_STEPS)
+
+    # Predict on held-out test data
+    test_input_fn = tf.estimator.inputs.numpy_input_fn(
+      x={WORDS_FEATURE: x_test},
+      y=y_test,
+      num_epochs=1,     # Note: For evaluation and prediction set this to 1,
+                        # so the input_fn will iterate over the data once and
+                        # then raise OutOfRangeError
+      shuffle=False)
+
+    predicted_test = classifier.predict(input_fn=test_input_fn)
+    test_out = pd.DataFrame(
+      [(p['classes'], p['probs'][1]) for p in predicted_test],
+      columns=['y_predicted', 'prob']
+    )
+    test_out['comment_text'] = x_train_text
+    test_out['y_true'] = y_test
+
+    # Write out predictions and probabilities for test data
+    tf.logging.info("Writing test predictions to {}".format(TEST_OUT_PATH))
+    test_out.to_csv(TEST_OUT_PATH)
+
+    # Score with sklearn and TensorFlow (hopefully they're the same!)
+    sklearn_score = metrics.accuracy_score(y_test, test_out['y_predicted'])
+    tf_scores = classifier.evaluate(input_fn=test_input_fn)
+
+    tf.logging.info('')
+    tf.logging.info('----------Evaluation on Held-Out Data---------')
+    tf.logging.info('Accuracy (sklearn)\t: {0:f}'.format(sklearn_score))
+    tf.logging.info('Accuracy (tensorflow)\t: {0:f}'.format(tf_scores['accuracy']))
+    tf.logging.info('')
+
+    # If specified, predict on unlabeled data
+    if FLAGS.predict_data is None:
+      return
+
+    data_unlabeled = WikiData(FLAGS.predict_data).data
```

Comment:
I went back and forth on this. I think you're right that now it doesn't add much as is. My thought was that in the future, if I want to do some preprocessing of the comment text and I want guarantee that I do the same thing at train and test time, it would be nice to encapsulate those steps in a class that represents collections of examples of wikipedia data.  I'd like to keep it this way for now, but if I end up not doing any pre-processing, I'll takeout the WikiData class because it is unnecessarily verbose.